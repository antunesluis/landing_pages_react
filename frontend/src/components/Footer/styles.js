import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    > ${SectionContainer} {
      padding: 0;
    }
  `}
`;
