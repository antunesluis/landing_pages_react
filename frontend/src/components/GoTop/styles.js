import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border-radius: 0.4rem;
    transition: all 0.3s ease-in-out;

    &:hover,
    &:focus {
      background: ${theme.colors.secondaryColor};
      color: ${theme.colors.white};
      transform: scale(1.05);
    }
  `}
`;
