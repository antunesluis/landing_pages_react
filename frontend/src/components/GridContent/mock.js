export default {
  title: 'Nossas Soluções Digitais',
  html: `
    <h2>Transforme seu Negócio com Nossas Soluções</h2>
    <p>
      Somos especialistas em desenvolver soluções <strong>inovadoras</strong> e <strong>personalizadas</strong> 
      para empresas que buscam se destacar no mercado digital. Nossa equipe combina expertise técnica 
      com visão estratégica para entregar resultados excepcionais.
    </p>

    <h3>Por que escolher nossa empresa?</h3>

    <ul>
      <li>
        <strong>Experiência Comprovada</strong> - Mais de 10 anos transformando ideias em 
        soluções digitais de sucesso
      </li>
      <li>
        <strong>Tecnologia de Ponta</strong> - Utilizamos as mais recentes tecnologias 
        do mercado para garantir performance e segurança
      </li>
      <li>
        <strong>Suporte Dedicado</strong> - Equipe sempre disponível para auxiliar 
        em todas as etapas do projeto
      </li>
    </ul>

    <blockquote>
      "A parceria com a empresa foi fundamental para o crescimento digital do nosso negócio. 
      A equipe é extremamente profissional e dedicada."
      <cite>- Maria Silva, CEO da TechCorp</cite>
    </blockquote>

    <h3>Nossos Serviços</h3>

    <p>
      Oferecemos uma gama completa de serviços digitais para impulsionar seu negócio:
    </p>

    <table>
      <thead>
        <tr>
          <th>Serviço</th>
          <th>Benefícios</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Desenvolvimento Web</td>
          <td>Sites responsivos, rápidos e otimizados para SEO</td>
        </tr>
        <tr>
          <td>Marketing Digital</td>
          <td>Estratégias personalizadas para aumentar sua presença online</td>
        </tr>
        <tr>
          <td>E-commerce</td>
          <td>Plataformas robustas e seguras para vendas online</td>
        </tr>
      </tbody>
    </table>

    <h3>Tecnologias que utilizamos</h3>
    <p>
      Trabalhamos com as tecnologias mais modernas do mercado, incluindo 
      <code>React</code>, <code>Node.js</code> e <code>Python</code>. 
      Nossa stack é constantemente atualizada para garantir as melhores soluções.
    </p>

    <h3>Entre em contato</h3>
    <p>
      Quer saber mais sobre como podemos ajudar seu negócio? 
      <a href="#contato">Clique aqui</a> para falar com um de nossos consultores ou 
      envie um email para <a href="mailto:contato@empresa.com">contato@empresa.com</a>
    </p>
  `,
};
