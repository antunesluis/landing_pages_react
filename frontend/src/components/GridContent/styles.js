import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const HtmlContent = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;

    // Estilização para textos e parágrafos
    p {
      font-size: ${theme.fonts.sizes.medium};
      line-height: 1.5;
      margin: ${theme.spacings.medium} 0;
    }

    // Estilização para cabeçalhos dentro do conteúdo
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      font-weight: 700;
      font-family: ${theme.fonts.family.secondary};
    }

    h1 {
      font-size: ${theme.fonts.sizes.xlarge};
      margin: ${theme.spacings.xlarge} 0;
    }

    h2 {
      font-size: ${theme.fonts.sizes.large};
      margin: ${theme.spacings.large} 0 ${theme.spacings.medium};
    }

    h3 {
      font-size: ${theme.fonts.sizes.medium};
      margin: ${theme.spacings.xlarge} 0 ${theme.spacings.small};
    }

    // Estilização para links
    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    // Estilização para listas
    ul,
    ol {
      margin: ${theme.spacings.small} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
    }

    ul {
      list-style: disc;
    }

    ol {
      list-style: decimal;
    }

    li {
      margin: ${theme.spacings.small} 0;
      font-size: ${theme.fonts.sizes.medium};
      text-align: left;
    }

    // Estilização para imagens
    img {
      max-width: 100%;
      height: auto;
    }

    // Estilização para citações
    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      color: ${theme.colors.primaryColor};
      margin: ${theme.spacings.medium} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
      font-style: italic;
    }

    // Estilização para tabelas
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 1px solid ${theme.colors.mediumGray};
      text-align: left;
    }

    table th {
      background: ${theme.colors.primaryColor};
      color: ${theme.colors.white};
    }

    // Estilização para elementos de ênfase
    strong,
    b {
      font-weight: bold;
    }

    em,
    i {
      font-style: italic;
    }

    /*
    @media ${theme.media.lteMedium} {
      h1 {
        font-size: ${theme.fonts.sizes.large};
      }
      h2 {
        font-size: ${theme.fonts.sizes.medium};
      }
      h3 {
        font-size: ${theme.fonts.sizes.small};
      }
      padding: ${theme.spacings.medium};
    } */
  `}
`;
