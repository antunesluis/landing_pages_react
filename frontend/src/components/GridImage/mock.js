export default {
  background: false,
  title: 'Galeria de projetos',
  description:
    'Conheça alguns dos projetos transformadores que desenvolvemos para nossos clientes. Cada solução é única e pensada para maximizar resultados.',
  grid: [
    {
      altText: 'E-commerce Platform',
      srcImg: 'https://picsum.photos/360/360?r=1',
      description:
        'Plataforma completa de e-commerce com mais de 1 milhão de usuários ativos',
    },
    {
      altText: 'Banking App Interface',
      srcImg: 'https://picsum.photos/360/360?r=2',
      description:
        'Interface moderna para app bancário com foco em experiência do usuário',
    },
    {
      altText: 'Healthcare Dashboard',
      srcImg: 'https://picsum.photos/360/360?r=3',
      description: 'Dashboard analítico para gestão de dados hospitalares',
    },
    {
      altText: 'Education Platform',
      srcImg: 'https://picsum.photos/360/360?r=4',
      description: 'Plataforma de ensino online com recursos interativos',
    },
    {
      altText: 'Real Estate Solution',
      srcImg: 'https://picsum.photos/360/360?r=5',
      description: 'Sistema integrado para gestão imobiliária',
    },
    {
      altText: 'Logistics System',
      srcImg: 'https://picsum.photos/360/360?r=6',
      description: 'Sistema de rastreamento e otimização logística',
    },
  ],
};
