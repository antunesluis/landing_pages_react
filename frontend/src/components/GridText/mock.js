export default {
  background: true, // Ativa o background com a imagem definida no tema
  title: 'Nossos Serviços',
  description:
    'Oferecemos soluções completas em desenvolvimento de software, com foco em qualidade e inovação. Nossa equipe especializada está pronta para transformar suas ideias em realidade.',
  grid: [
    {
      title: 'Software Architecture',
      description:
        'Planejamos e implementamos arquiteturas robustas e escaláveis, seguindo as melhores práticas do mercado. Utilizamos padrões de projeto modernos.',
    },
    {
      title: 'Web Development',
      description:
        'Criamos aplicações web modernas e responsivas utilizando as mais recentes tecnologias do mercado. Nosso foco está em React, Next.js e Node.js.',
    },
    {
      title: 'UX/UI Design',
      description:
        'Nossa equipe de design cria interfaces intuitivas e atraentes, focando na experiência do usuário. Trabalhamos com prototipação, testes de usabilidade e design systems completos.',
    },
    {
      title: 'DevOps & Cloud',
      description:
        'Implementamos práticas de CI/CD e gerenciamos toda infraestrutura em cloud, utilizando AWS, Azure ou Google Cloud. Garantimos a disponibilidade e segurança da sua aplicação.',
    },
    {
      title: 'Mobile Development',
      description:
        'Desenvolvemos aplicativos mobile nativos e híbridos para iOS e Android. Utilizamos React Native e Flutter para criar experiências móveis excepcionais com código compartilhado.',
    },
    {
      title: 'Consultoria Técnica',
      description:
        'Oferecemos consultoria especializada em arquitetura de software, escolha de tecnologias e melhores práticas de desenvolvimento. Ajudamos sua equipe a tomar as melhores decisões técnicas.',
    },
  ],
};
