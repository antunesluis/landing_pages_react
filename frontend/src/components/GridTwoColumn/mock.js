export default {
  title: 'Grid two columns',
  text: 'To wrap up January, we are proud to bring you the release of Firefox 85. In this version we are bringing you support for the :focus-visible pseudo-class in CSS and associated devtools, and the complete removal of Flash support from Firefox. ',
  srcImg: 'assets/images/illustration/programador.svg',
};
