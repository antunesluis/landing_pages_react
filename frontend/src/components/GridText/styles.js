import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      max-width: 70ch;
      opacity: 0.9;
    }

    > ${HeadingContainer} {
      text-shadow: 0.1rem 0.1rem 0.2rem rgba(0, 0, 0, 0.2);
      margin-bottom: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      ${TextComponent} {
        margin-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    max-width: 100%;
    grid-template-columns: repeat(auto-fit, minmax(min(100%, 32rem), 1fr));
    gap: ${theme.spacings.xlarge};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    border-radius: 8px;
    background: rgba(255, 255, 255, 0.03);
    transition: all 0.3s ease-in-out;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-6px);
      background: rgba(255, 255, 255, 0.05);
    }

    ${HeadingContainer} {
      position: relative;
      left: 5rem;
      margin-bottom: ${theme.spacings.small};

      // Evita que o texto encoste na lateral direita
      padding-right: ${theme.spacings.medium};
      // Garante que o heading n√£o ultrapasse o container
      max-width: calc(100% - 5rem);

      @media ${theme.media.lteMedium} {
        left: 4rem;
        max-width: calc(100% - 4rem);
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: clamp(5rem, 8vw, 7.5rem);
      font-weight: 700;
      opacity: 0.15;
      top: -3rem;
      left: -6rem;
      transform: rotate(8deg);
      transition: all 0.2s ease-in-out;
    }

    &:hover ${HeadingContainer}::before {
      opacity: 0.2;
      transform: rotate(5deg) scale(1.1);
    }

    ${TextComponent} {
      line-height: 1.6;
      opacity: 0.85;
      padding-right: ${theme.spacings.small};
      max-width: 100%;
    }
  `}
`;
