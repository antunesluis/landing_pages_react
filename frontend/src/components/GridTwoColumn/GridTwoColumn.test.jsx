import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { GridTwoColumn } from '.';

import mock from './mock';

describe('<GridTwoColumn />', () => {
  it('should render', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
