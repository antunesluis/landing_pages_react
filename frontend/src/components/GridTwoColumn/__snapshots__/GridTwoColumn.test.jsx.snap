// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: minmax(280px, 1.2fr) 1.3fr;
  align-items: center;
  gap: 4.0rem;
  margin: 0 auto;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

.c3 {
  order: 1;
  max-width: 60ch;
}

.c7 {
  order: 2;
  width: 100%;
  margin: 0 auto;
  padding: 1.6rem;
}

.c8 {
  width: 100%;
  height: auto;
  object-fit: cover;
  transition: all 300ms ease-in-out;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width: 768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
    gap: 1.6rem;
  }
}

@media (max-width: 768px) {
  .c3 {
    margin-bottom: 2.4rem;
    order: 2;
    max-width: 100%;
  }
}

@media (max-width: 1170px) {
  .c7 {
    max-width: 600px;
  }
}

@media (max-width: 768px) {
  .c7 {
    order: 1;
    margin-bottom: 2.4rem;
    max-width: 500px;
  }
}

@media (max-width: 768px) {
  .c8 {
    max-height: 40rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid two columns
          </h2>
          <p
            class="c6"
          >
            To wrap up January, we are proud to bring you the release of Firefox 85. In this version we are bringing you support for the :focus-visible pseudo-class in CSS and associated devtools, and the complete removal of Flash support from Firefox. 
          </p>
        </div>
        <div
          class="c7"
        >
          <img
            alt="Grid two columns"
            class="c8"
            src="assets/images/illustration/programador.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
